client/src/modules/shopping-list/model/reducer.js

```diff
@@ -67,7 +68,8 @@ const lists = (state = {}, action) => {
     }
     case ListActionTypes.ARCHIVE_SUCCESS: {
       const { listId: _id, isArchived } = action.payload;
-      const archivedList = { _id, isArchived };
+      const { name } = state[action.payload.listId];
+      const archivedList = { _id, isArchived, name };
```

Comment:
here you can do the same const prevList = state[action.payload.listId]

```
return {
...state,
[action.payload.listId]: {...prevList, isArchived}
}
```

Reply:
I think we can't do this because we want to remove all data we don't need except { _id, isArchived, name }, we remove permissions, description, products(items) etc. ... Your proposal is much more cleaner but don't remove unnecessary data...